src/data/complex/is_R_or_C.lean

```diff
@@ -58,18 +61,24 @@ noncomputable abbreviation of_real (r : ℝ) := algebra_map ℝ K r
 local notation `𝓚` := @is_R_or_C.of_real K _
 local postfix `†`:100 := @is_R_or_C.conj K _
 
-lemma of_real_alg (x : ℝ) : 𝓚 x = x • (1 : K) :=
-algebra.algebra_map_eq_smul_one x
+-- see Note [coercion into rings]
+@[priority 900] noncomputable instance cast_coe : has_coe_t ℝ K := ⟨of_real⟩
+
+lemma coe_eq_of_real {x : ℝ} : @is_R_or_C.of_real K _ x = x := rfl
```

Comment:
This should be a simp lemma: you have a normal form, which is using the coercion, so you want `simp` to convert all other forms to this normal form.

Reply:
If coercion is the normal form, and `algebra_map` is the actual implementation, is there any point keeping the intermediate `of_real` definition?

Reply:
Good point. I removed it.